models/complaints.js

```diff
@@ -2,24 +2,27 @@ var mongoose = require('mongoose');
 var User = require('./user');
 var complaintSchema = new mongoose.Schema(
     {
-        name: { type: String, required: true },
-        desc: { type: String },
+        name: String,
+        desc: String,
+        createdAt: { type: Date, default: Date.now },
         reviewStartedAt: { type: Date },
+        reviewPendingAt: { type: Date },
+        reviewReopenAt: { type: Date },
+        reviewClosedAt: { type: Date },
         author: {
-            type: mongoose.Schema.Types.ObjectId,
-            ref: 'User',
+            id: {
+                type: mongoose.Schema.Types.ObjectId,
+                ref: 'User',
+            },
         },
         archived: { type: Boolean, default: false },
         archivingTime: { type: Date },
         status: {
             type: String,
-        } /** Pending => createdAt
-        Open => reviewStartedAt
-        Close => updatedAt */,
-        assignedTo: {
-            type: mongoose.Schema.Types.ObjectId,
-            ref: 'User',
         },
+        assignedTo: String,
+        responseTime: String,
+        resolveTime: String,
```

Comment:
don't store in db dynamically calculated data. it's bad practice. i coud show you in action. what problems it caused me in production with Rolling Records.